pkg/controller/util/endpointslice/cache.go

```diff
@@ -106,21 +110,27 @@ func (c *Cache) UpdateEndpointPortCache(endpointsNN types.NamespacedName, epCach
 	// endpoints in the provided spCache and any spCache it might be replacing.
 	c.numEndpoints = c.numEndpoints + currNumEndpoints - prevNumEndpoints
 
-	c.cache[endpointsNN] = epCache
-	c.updateMetrics()
+	c.cache[portNN] = epCache
+	mUpdate := c.desiredAndActualSlices()
+	return mUpdate.actual, mUpdate.desired, c.numEndpoints
 }
 
 // DeleteEndpoints removes references to an Endpoints resource from the global
 // cache and updates the corresponding metrics.
-func (c *Cache) DeleteEndpoints(endpointsNN types.NamespacedName) {
+func (c *Cache) DeleteEndpoints(portNN types.NamespacedName) (epExist bool, actual, desired, numEndpoints int) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
```

Comment:
This change is hard to understand without context, maybe you should use different commits for different changes, at least I´m having problems to follow the logic in this metrics cache changes